driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/BatchCursor.java

```diff
@@ -38,6 +39,21 @@ public Publisher<List<T>> next() {
         return Mono.create(sink -> wrapped.next(sinkToCallback(sink)));
     }
 
+    public Publisher<List<T>> nextWithSink(final FluxSink<T> parentSink) {
```

Comment:
No way to just have a single method here?  It could cause future bugs if callers are confused about which one to use.

Reply:
One thought I had is to add the parent FluxSink as a field, but I'm not sure how we'd get it through to the constructor.

Reply:
I'll investigate - I think cursor.first() doesn't use a parent sink to handle the flattening of the batches so doesn't follow the same pattern - but will check.

Reply:
Turns out Mono Sink and Flux Sink aren't the same - so couldnt remove both methods. I did neaten it up though with a Supplier instead.